### ISSUE TITLE
`:redir` causes shell prompt to move when exiting vim

### ISSUE THREAD

**Issue by itspriddle** [tpope/vim-sensible#182]
Since cdb3801 was added I have noticed that exiting Vim causes my shell prompt to move to the bottom of my terminal window.

I've whittled it down to the following case to reproduce:

```vim
" test.vim
set nocompatible
source vim-sensible/plugin/sensible.vim
```

On dad453e (master as of today):

<img width="1459" alt="Screenshot 2023-01-03 at 10 21 52 AM" src="https://user-images.githubusercontent.com/49571/210387089-9c54bdfb-5c3e-4511-aa48-e7b8190c6b87.png">

On c597927:

<img width="1459" alt="Screenshot 2023-01-03 at 10 22 07 AM" src="https://user-images.githubusercontent.com/49571/210387209-1ca46626-f6ca-4c4a-b21c-ac5227347df2.png">

It seems this might be an upstream vim issue as it can be duplicated without sensible:

```
set nocompatible
redir => out
silent verbose exe 'setglobal backspace?'
redir END
```

Any thoughts on how this could be avoided? Thanks! 

---

```
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Dec 26 2022 13:51:26)
macOS version - arm64
Included patches: 1-1100
```

**Comment by itspriddle** [comment]
8e274c2 doesn't fix it I'm afraid.

It might actually be something with `set ?` and friends, not `redir`.

This will also reproduce it:

```
" test.vim
set nocompatible
let out = execute('verbose setglobal number?')
```

Then in a terminal:

```
clear && vim -u test.vim -c q
```

The key thing to reproducing it is you have to have blank space between your shell prompt and the bottom of the terminal window.

The terminal doesn't seem to matter. I've tried it in latest macOS/Terminal.app/iTerm2. Screen/tmux show the same behavior.

I have a few different vim versions available and they show the same going back to 8.1. 

Definitely a weird one. I'm happy to provide any other info or tests that might help!

**Comment by pwsiegel** [comment]
I have confirmed that https://github.com/tpope/vim-sensible/commit/947f613a091d93316c47c0249237f4529fef5182 fixed the problem on my machine. Thank you for the quick response, and for your years of hard work making vim a great tool!
